lib/Compiler.cpp

```diff
@@ -151,176 +157,192 @@ static llvm::cl::opt<bool> verify("verify", llvm::cl::init(false),
 
 // Populates |SamplerMapEntries| with data from the input sampler map. Returns 0
 // if successful.
-int ParseSamplerMap(llvm::SmallVectorImpl<std::pair<unsigned, std::string>> *SamplerMapEntries) {
-  if (!SamplerMap.empty()) {
+int ParseSamplerMap(
+    const std::string &sampler_map,
+    llvm::SmallVectorImpl<std::pair<unsigned, std::string>> *SamplerMapEntries,
+    llvm::raw_ostream *error_str) {
```

Comment:
Passing by it by reference would make it easier to use.

Reply:
[Google style guide](https://google.github.io/styleguide/cppguide.html) says all references must be const-qualified.